pyaedt/desktop.py

```diff
@@ -1524,3 +1536,34 @@ def install(package_path, package_name=None):
         if aedt_version >= "2023.2":
             write_toolkit_config(os.path.join(toolkit_dir, product), lib_dir, toolkit_name, toolkit=toolkit)
         self.logger.info("{} toolkit installed.".format(toolkit_name))
+
+    @pyaedt_function_handler()
+    def _launch_aedt(self, full_path, non_graphical, port, first_run=True):
+        """Launch AEDT in gRPC mode."""
+
+        def launch_desktop_on_port():
+            command = [full_path, "-grpcsrv", str(port)]
+            if non_graphical:
+                command.append("-ng")
+            my_env = os.environ.copy()
+            for env, val in settings.aedt_environment_variables:
+                my_env[env] = val
+            if is_linux:
+                subprocess.Popen(command, env=my_env, stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
+            else:
+                subprocess.Popen(" ".join(command), env=my_env, stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
+
+        self._aedt_process_thread = threading.Thread(target=launch_desktop_on_port)
+        self._aedt_process_thread.daemon = True
+        self._aedt_process_thread.start()
+        timeout = 180
```

Comment:
Maybe 180 s is too long for the timeout. What do you think?

Reply:
in windows typically is true. On linux it depends if there are deps check

Reply:
moved to settings.desktop_launch_timeout variable